### ISSUE TITLE
Automatic reminder for daily log

### ISSUE THREAD

**Issue by anikalindtner** [rails-girls-summer-of-code/rgsoc-teams#256]
As we talked about in our weekly organizer hangout yesterday, a lot of what the supervisors are doing is nagging teams to get stuff done. Which wears them down and shouldn't be the main task of our supervisors. So I thought about how to make at least *some* of the process automated to take it out of their hands. And one thing would be to send an email reminder at hour xy to the team if they haven't filled out their daily log. This way we nag them without having to do it ourselves and they get accustomed to doing it everyday from the beginning on. 

Exception would have to be weekends. 

@sareg0 was interested in helping to think about how we could implement something like that. 

Comments? Opinions?

**Comment by carpodaster** [comment]
Yeah, we have talked about this enhancement to the minilog feature before and I think it's a good one.

Possible gotcha: The reminder should take the team's timezone into account.

**Comment by alicetragedy** [comment]
@carpodaster yep, I mentioned the thing with the timezone briefly during the call when this was discussed :) but I also think that it would be a great way to annoy everyone to have them do their log so +1 on this

**Comment by idengager** [comment]
This is a very good idea. :+1:

**Comment by sareg0** [comment]
yeah, I really like this idea :) +1 from me as possible new feature. I think it would be super cool to work on, but if it happens to be something the team would like to work on, that's also cool :)

**Comment by F3PiX** [comment]
I was thinking: with the new Supervisor Dashboard, it would be quite easy to add an automatic mailer function/button. The supervisor would be the one to initiate the action, and the mail itself could be sent with the orga-team as the sender. Ideally, it would count the warnings as well.
Wdyt?

**Comment by sareg0** [comment]
it sounds interesting for sure, particularly since the supervisor can have a bit of a running record of how many reminders have been sent.
Re. the warnings though, the orga team makes a decision on sending those, so it feels like a different concern to being in the  supervisor's dashboard.

**Comment by F3PiX** [comment]
Aha, I see: there are reminders and there are warnings. So only the reminder should be sent by the supervisor, cc orga.

**Comment by anikalindtner** [comment]
@michaelem and i will try to tackle this. would be helpful to have a data dump for that. @alicetragedy could you help on this?

**Comment by michaelem** [comment]
I would put this in a rake task that could run once a day and checks all the teams, sends emails to those who did not put out an update. Do we already have any tasks that run on a schedule?

Stackoverflow says we should use this: https://elements.heroku.com/addons/scheduler
I have no idea how these things work in heroku, can someone help me with the scheduling part?

**Comment by klappradla** [comment]
You can ping me once you're there, I've already used this add-on (it's quite simple).

**Comment by alicetragedy** [comment]
@anikalindtner @michaelem I will do a new data dump and send it to you :) As far as I know we don't have any tasks that run on a schedule, but I think Carsten might be the most fitting person to answer this question.
Something else that comes to mind: if we want notifications to work for all participants, we probably have to make sure we have all of their email addresses stored + make email a requirement also (right now it's just pulled from GitHub with the OAuth, so we only have those of addresses that are public) — related issue: https://github.com/rails-girls-summer-of-code/rgsoc-teams/issues/413

**Comment by carpodaster** [comment]
@michaelem we don't have any recurring tasks set up at the moment, no. But as @klappradla said, that's no problem.

**Comment by carpodaster** [comment]
Haha, scratch that. _Of course_ we already have one set up. Just ping me with the rake command that needs to be added once the feature is merged into master.

**Comment by michaelem** [comment]
Now that #475 is merged we need configure heroku to run teams:notify_missing_log_updates once per day, I'd say at 23:59 utc if possible?

**Comment by carpodaster** [comment]
<img width="772" alt="bildschirmfoto 2016-07-02 um 11 14 53" src="https://cloud.githubusercontent.com/assets/164400/16539527/43afa45e-4046-11e6-8c78-9ddc9c5ee9a2.png">

rake task is scheduled now. 23:30 is the closest I can get to 23:59 ;)

**Comment by carpodaster** [comment]
@anikalindtner @michaelem can this be closed?

**Comment by carpodaster** [comment]
Closing. I believe this is done. Please re-open if I'm mistaken.
